rockcraft/extensions/gunicorn.py

```diff
@@ -113,27 +114,18 @@ def _gen_parts(self) -> dict:
         return parts
 
     def _check_async(self) -> str:
-        """Check if gevent package installed in requirements.txt.
-
-        Regex will match with exact `gevent` package and also
-        will match `gevent` with versions such as:
-          - gevent==*.*
-          - gevent ==0.*
-          - gevent <2.*
-          - gevent<2.*
-          - gevent >2.*
-          - gevent>2.*
-          - gevent@github.com/*
-          - gevent[some_extra]
-          - gevent [some_extra]
-        """
+        """Check if gevent package installed in requirements.txt."""
         requirements_file = self.project_root / "requirements.txt"
-        with requirements_file.open() as f:
-            lines = f.readlines()
-            for line in lines:
-                for token in line.split():
-                    if re.match("^gevent[|[=|@|\\s|\n<|>]|(^gevent$)", token):
-                        return "gevent"
+        requirements_text = requirements_file.read_text()
+        for line in requirements_text.splitlines():
+            if not line.strip():
+                continue
+            clean_line = line.strip()
+            if clean_line.startswith("#"):
+                continue
+            req = Requirement(clean_line)
```

Comment:
What happens if we skip the two checks? Do we get an exception?

Reply:
Nope, it means `gevent` package is not in the `requirements.txt` file so the rock is sync.

Reply:
I mean what happens if you delete lines 121-125, what happens if we get an empty line or a line starting with #?

Reply:
I'm thinkking we can just try catch pass rather than have these lines

Reply:
Also, I would move line 123 above 121 and just check if the clean line is empty

Reply:
If I remove we get an error. Sure, we can do try/catch here.